Keep downloads during sync process for unsubscribed podcasts

bjtitus opened the pull request:
Fixes [PCIOS-408](https://linear.app/a8c/issue/PCIOS-408/downloaded-episodes-from-unfollowed-podcasts-are-removed-from)

This change was made in #3761 in order to remove downloads of unsubscribed podcasts, including unsubscribes synced from the server. However, we don't want to remove downloads for episodes downloaded separately from subscriptions even though the corresponding podcast is cached in the podcast table.

## To test

* Download an episode by browsing an unsubscribed show in Discover
* ✅ Ensure the download completes
* Refresh in Profile
* ✅ Ensure the download remains

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

bjtitus committed:
Keep downloads removed during sync process

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is assigned to the mil...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/Automattic/pocket-casts-ios/milestone/111">8.2 ❄️</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

bjtitus committed:
Add unit test to cover this case

We might change this in the future to consider podcasts unsubscribed from another device.

danielebogo reviewed (approved):
@bjtitus following you steps I downloaded episodes from unfollowed Podcasts from discover and all those are still in my Downloaded section after reloading the Profile
